pkg/security/api/v1/types.go

```diff
@@ -80,16 +80,17 @@ type PodSecurityPolicyReview struct {
 // PodSecurityPolicyReviewSpec defines specification for PodSecurityPolicyReview
 type PodSecurityPolicyReviewSpec struct {
 	// template is the PodTemplateSpec to check. The template.spec.serviceAccountName field is used
-	// if serviceAccountNames is empty, unless the template.spec.serviceAccountName is empty,
+	// if serviceAccountNames is not nil and empty, unless the template.spec.serviceAccountName is empty,
 	// in which case "default" is used.
-	// If serviceAccountNames is specified, template.spec.serviceAccountName is ignored.
+	// If serviceAccountNames is specified or nil, template.spec.serviceAccountName is ignored.
```

Comment:
the idea that an explicit list of names overrides what's in the template makes sense to me

nil meaning "override to all" and empty list meaning "use what's in the template" is odd to me. do you expect it to be common for someone to use this to try to troll for some service account in the namespace that has the permissions they expect? I'd expect to use service accounts per app, and use this to check if the service account has the required permissions. picking one out of the list of all service accounts just because it has the right scc permissions seems like a bad approach to selecting a service account